CONTRIBUTING.md

```diff
@@ -63,14 +63,14 @@ Ideally, flake and ufmt should be run via pre-commit hooks.
 But if for some reason you want to run them separately follow this:
 
 ```
+pip install -r requirements-lint.txt
```

Comment:
Just wondering, do we really need a separate requirements file just for lint? I don't really see this in other repos (besides pytorch core, which is a much larger repo anyways) and wonder if we're adding unnecessary files at the top-level of the repo by doing this

Reply:
I suppose other repo uses just a vanilla flake8 that does not involve plug-ins to be installed. If the latter, the requirement file is just to consolidate all the plug-ins in one place.

Reply:
Also, as long as we don't specify it in `setup.py`, the additional top-level requirement files won't go into installation or packaging.

Reply:
That's reasonable. Maybe I can instead ask a strictly stronger question: do we really need to allow users to run the lint suite manually 🙂 ?  Basically my concern is that we will give the appearance of extra friction in development by adding this file, which we do not want.

Reply:
Manual, you mean running lint on the command line? The developer can also use the correct version of flake8 inside their IDE. 

Where is the extra friction though? If I want to lint on my own, I need to know what the library uses.

Reply:
> Manual, you mean running lint on the command line?

Yes, by following the instructions we provide in CONTRIBUTING.

> Where is the extra friction though?

So it is not actually extra friction. But it gives the _appearance_ of extra friction. We will now have three separate files all containing "requirements" in their names, all of which are present on the GitHub landing page. Just from a first-time user's perspective, if I engage in the usual exercise of looking at various files to better understand the repo and I see this many different types of requirements I would feel a bit daunted. And I think the current complexity/maturity of our repo is not high enough to justify a bunch of disjoint sets of requirements like this.